modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * observability/otel/otel-troubleshooting.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="troubleshooting-instrumentation_{context}"]
+= Troubleshooting instrumentation
+
+You can troubleshoot instrumentation issues as follows.
+
+.Procedure
+
+== Ensure the Instrumentation was injected properly in your workload
+
+. Check the instrumentation object is created:
++
+[source,console]
+----
+oc get instrumentation -n <your-project>
+----
+
+. Verify that the resources were deployed in the correct order
++
+Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.
+
+When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
```

Comment:
What is `your-project`? Is this an OpenShift namespace or what?

Reply:
Yes, it is a namespace. In OpenShift, namespaces are called `Projects`. But they can be interchanged.

Reply:
And what does `your` mean?

Reply:
It means "the namespace where your stuff is deployed". Like place <here> your stuff.

Reply:
Yes, but they probably use multiple custom namespaces. How would you describe the namespace that is relevant here?

Reply:
Oh ok. It would be the namespace where the workloads (that expect the instrumentation to be injected) are located